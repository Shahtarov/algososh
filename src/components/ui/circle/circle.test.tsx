import { render, screen } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Circle component", () => {
	it("text inside", () => {
		render(<Circle letter={"text"} />);

		const circle = screen.getByTestId("circle");
		const textInCircle = screen.getByTestId("textInside");

		expect(textInCircle).toHaveTextContent(/text/i);
		expect(circle).toMatchSnapshot();
	});

	it("without text inside", () => {
		render(<Circle />);

		const circle = screen.getByTestId("circle");
		const textInCircle = screen.getByTestId("textInside");

		expect(textInCircle).toHaveTextContent("");
		expect(circle).toMatchSnapshot();
	});

	it("text in head", () => {
		render(<Circle head="text" />);

		const circle = screen.getByTestId("circle");
		const textInHead = screen.getByTestId("head");

		expect(textInHead).toHaveTextContent(/text/i);
		expect(circle).toMatchSnapshot();
	});

	it("react element in head", () => {
		render(<Circle head={<div data-testid="inner-element" />} />);

		const circle = screen.getByTestId("circle");
		const innerElement = screen.getByTestId("inner-element");

		expect(innerElement).toBeInTheDocument();
		expect(circle).toMatchSnapshot();
	});

	it("text in tail", () => {
		render(<Circle tail="text" />);

		const circle = screen.getByTestId("circle");
		const textInTail = screen.getByTestId("tail");

		expect(textInTail).toHaveTextContent(/text/i);
		expect(circle).toMatchSnapshot();
	});

	it("number in index", () => {
		render(<Circle index={10} />);

		const circle = screen.getByTestId("circle");
		const index = screen.getByTestId("index");

		expect(index).toHaveTextContent(/10/);
		expect(circle).toMatchSnapshot();
	});

	it("is small", () => {
		render(<Circle isSmall={true} />);

		const circle = screen.getByTestId("circle");
		const state = screen.getByTestId("state");

		expect(state).toBeInTheDocument();
		expect(state).toHaveClass("small");
		expect(circle).toMatchSnapshot();
	});

	it("is default", () => {
		render(<Circle state={ElementStates.Default} />);

		const circle = screen.getByTestId("circle");
		const state = screen.getByTestId("state");

		expect(state).toBeInTheDocument();
		expect(state).toHaveClass("default");
		expect(circle).toMatchSnapshot();
	});

	it("is changing", () => {
		render(<Circle state={ElementStates.Changing} />);

		const circle = screen.getByTestId("circle");
		const state = screen.getByTestId("state");

		expect(state).toBeInTheDocument();
		expect(state).toHaveClass("changing");
		expect(circle).toMatchSnapshot();
	});

	it("is modified", () => {
		render(<Circle state={ElementStates.Modified} />);

		const circle = screen.getByTestId("circle");
		const state = screen.getByTestId("state");

		expect(state).toBeInTheDocument();
		expect(state).toHaveClass("modified");
		expect(circle).toMatchSnapshot();
	});
});
