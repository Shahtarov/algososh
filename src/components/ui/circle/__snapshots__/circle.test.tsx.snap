// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component is changing 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   changing"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component is default 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component is modified 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   modified"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component is small 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle  small default"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component number in index 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  >
    10
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component react element in head 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  >
    <div
      data-testid="inner-element"
    />
  </div>
  <div
    class="circle   default"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component text in head 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head"
  >
    text
  </div>
  <div
    class="circle   default"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component text in tail 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail"
  >
    text
  </div>
</div>
`;

exports[`Circle component text inside 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    >
      text
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component without text inside 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="state"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="textInside"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
